engine/src/main/java/io/camunda/zeebe/engine/api/ProcessingResultBuilder.java

```diff
@@ -22,7 +22,8 @@ public interface ProcessingResultBuilder {
    * Appends a record to the result
    *
    * @return returns itself for method chaining
-   * @throws RuntimeException if to appended record doesn't fit into the RecordBatch
+   * @throws io.camunda.zeebe.engine.api.records.RecordBatch.ExceededBatchRecordSizeException if the
+   *     appended record doesn't fit into the RecordBatch
```

Comment:
👍 I like it

🔧 please also add this to the appendX methods of the different writers (stateWriter, rejectionWriter, etc). These are directly used by the processors so would add the visibility in the processors